src/deal_ii/src/4C_deal_ii_context.hpp

```diff
@@ -31,10 +59,150 @@ namespace DealiiWrappers
    * this class.
    */
   template <int dim, int spacedim = dim>
-  struct Context
+  class Context
+  {
```

Comment:
So you changed my original idea behind this class. I am not saying you shouldn't do this, but I wonder if it is really what we want. The fact that you have to make a function a `friend` seems to indicate that this is really just a collection of data, that some functions can operate on.